src/main/java/cheese/Storage.java

```diff
@@ -0,0 +1,106 @@
+package cheese;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+import cheese.task.Deadline;
+import cheese.task.Event;
+import cheese.task.Task;
+import cheese.task.ToDo;
+
+
+
+/**
+ * Manipulate file that store tasks information
+ */
+public class Storage {
+    private final String filePath;
+
+    Storage(String filePath) {
+        this.filePath = filePath;
+    }
+
+    /**
+     * Return or create list file
+     * @return File
+     * @throws IOException In case of IOException
+     */
+    private File getListFile() throws IOException {
+        File f = new File(filePath);
+        if (!f.exists()) {
+            boolean created = f.createNewFile();
+            System.out.println(created);
+        }
+        return f;
+    }
+
+    /**
+     * Load tasks from file
+     * @return tasks from file
+     * @throws CheeseException I/O exception or Incorrect data format
+     */
+    public ArrayList<Task> loadTasks() throws CheeseException {
+        ArrayList<Task> tasks = new ArrayList<>();
+        try {
+            File f = getListFile();
+            Scanner s = new Scanner(f);
+            while (s.hasNext()) {
+                String data = s.nextLine();
+                tasks.add(loadTask(data));
+            }
+        } catch (IOException | CheeseException e) {
+            throw new CheeseException("Load tasks failed." + e.getMessage());
+        }
+        return tasks;
+    }
+    private Task loadTask(String s) throws CheeseException {
+        String[] data = s.split(",");
+        return switch (data[0]) {
+        case "T" -> new ToDo(data);
+        case "D" -> new Deadline(data);
+        case "E" -> new Event(data);
+        default -> throw new CheeseException("Incorrect data format");
+        };
+    }
+
+    /**
+     * Add and save new task
+     * @param t task to be added
+     */
+    public void add(Task t) throws CheeseException {
```

Comment:
Could the naming appear a little inconsistent? The previous method was called loadTask while the following are just add and update rather than addTask and updateTask